dali/operators/reader/numpy_reader_gpu_op.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_NUMPY_READER_GPU_OP_H_
+#define DALI_OPERATORS_READER_NUMPY_READER_GPU_OP_H_
+
+#include <utility>
+#include <string>
+#include <vector>
+
+#include "dali/operators/reader/reader_op.h"
+#include "dali/operators/reader/loader/numpy_loader_gpu.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/transpose/transpose_gpu.h"
+
+namespace dali {
+
+class NumpyReaderGPU : public DataReader<GPUBackend, ImageFileWrapperGPU> {
+ public:
+  explicit NumpyReaderGPU(const OpSpec& spec) :
+    DataReader<GPUBackend, ImageFileWrapperGPU>(spec),
+    thread_pool_(num_threads_, spec.GetArgument<int>("device_id"), false) {
+    prefetched_batch_tensors_.resize(prefetch_queue_depth_);
+
+    // set a device guard
+    DeviceGuard g(device_id_);
+
+    // init loader
+    bool shuffle_after_epoch = spec.GetArgument<bool>("shuffle_after_epoch");
+    loader_ = InitLoader<NumpyLoaderGPU>(spec, std::vector<string>(),
+                                         shuffle_after_epoch);
+
+    // init thread pool:
+    kmgr_.Resize<Kernel>(1, 1);
+  }
+
+  ~NumpyReaderGPU() {
+    StopPrefetchThread();
+    thread_pool_.WaitForWork();
```

Comment:
I think this should not be necessary since RunAll() already waits for the work to finish.

Reply:
Done